### PR TITLE
fix: lane resize constraint 

### LINKED ISSUE #2209: Lane resize constraints do not work when resizing from bottom right corner with any lane but the lowest one
### Describe the Bug

The lane resize constraints do not work when using any lane except for the lowest one of the participant and using the bottom right corner of the lane.

chrome_3YOVUbaBhp

### Steps to Reproduce

1. Create a pool with at least two lanes
2. Insert an element into one of the lanes
3. Resize any but the lowest lane with the direction "se"

Result: The width of the lane can be decreased beyond the elements inside of it.

### Expected Behavior

The participant and therefore one of the lanes should always contain all elements. It should not be possible to decrease the x coordinate of a lane/participant beyond that of the elements inside of it.

### Environment

 - Browser: Chrome v127.0.6533.73
 - OS: Windows 11 Enterprise
 - Library version: 17.6.4, also [demo](https://demo.bpmn.io/new)

### PR DESCRIPTION
**Pull request by abdul99ahad**
**Issue:** The lane resize constraints do not work when using any lane except for the lowest one of the participant and using the bottom right corner of the lane.

image

image

**Resolution:** Fixed lane resizing when multiple lanes exist with participants from NW and SE direction. 

image

**Identified Issue:** isFirst and isLast flag was not setting with corresponding lanes during resizing. 

Closes #2209

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ All notable changes to [bpmn-js](https://github.com/bpmn-io/bpmn-js) are documen
 
 ___Note:__ Yet to be released changes appear here._
 
+## 17.9.3
```
Comment (philippfromme) [inline CHANGELOG.md:9]: Let's not add the next version yet and keep this in the unreleased changes section. We don't know what exactly what the next version will be. It also might be a minor or major.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bpmn-js",
-  "version": "17.9.2",
```
Comment (philippfromme) [inline package.json:3]: Please remove this change. The version will be updated once we release. We will talk about this later. 😉

**Review by philippfromme — DISMISSED** [review]

**Comment by philippfromme** [comment]
As always make sure `Closes #xxx` is part of the commit message of the relevant commit.

**Comment by sombrek** [comment]
@abdul99ahad Nice work.

I've checked out your branch and noticed the fix does not yet apply to vertical pools. Also the vertical lane now enlarges when using the resize handle. This seems unexpected. Please have a look.
[sample.bpmn.txt](https://github.com/user-attachments/files/16627864/sample.bpmn.txt)

<details>
<summary>Visual demo</summary>

visual-demo
</details>

**Comment by philippfromme** [comment]
> @abdul99ahad Nice work.
> 
> I've checked out your branch and noticed the fix does not yet apply to vertical pools. Also the vertical lane now enlarges when using the resize handle. This seems unexpected. Please have a look. [sample.bpmn.txt](https://github.com/user-attachments/files/16627864/sample.bpmn.txt)
> 
> Visual demo

I found the same issue when reviewing this pull request. @abdul99ahad is working on fixing it.

**Comment by abdul99ahad** [comment]
> > @abdul99ahad Nice work.
> > I've checked out your branch and noticed the fix does not yet apply to vertical pools. Also the vertical lane now enlarges when using the resize handle. This seems unexpected. Please have a look. [sample.bpmn.txt](https://github.com/user-attachments/files/16627864/sample.bpmn.txt)
> > Visual demo
> 
> I found the same issue when reviewing this pull request. @abdul99ahad is working on fixing it.

I have fixed the issue for vertical lane handling.

**Comment by nikku** [comment]
Is this related to https://github.com/bpmn-io/bpmn-js/issues/2209? If so, please link it. Please also use the existing PR template (reminds you of linking) :slightly_smiling_face:

File: lib/features/modeling/behavior/ResizeBehavior.js [hunk 3]
```diff
@@ -160,10 +160,25 @@ export function getParticipantResizeConstraints(laneShape, resizeDirection, bala
...
+    // lane flags
+    if (isHorizontalLane) {
+      if (otherTrbl.top < (laneTrbl.top - 10)) {
+        isFirst = false;
+      }
+      if (otherTrbl.bottom > (laneTrbl.bottom + 10)) {
+        isLast = false;
+      }
+    }
+    else if (!isHorizontalLane) {
```
Comment (nikku) [inline lib/features/modeling/behavior/ResizeBehavior.js:172]: I believe this is duplicated, right? If one branch checks for `isHorizontalLane` then the other one naturally checks for the opposite (`!isHorizontalLane`)?

```suggestion
    else {
```

File: test/spec/features/modeling/behavior/ResizeBehaviorSpec.js [hunk 4]
```diff
@@ -326,6 +326,63 @@ describe('features/modeling - resize behavior', function() {
         })
       );
```
Comment (nikku) [inline test/spec/features/modeling/behavior/ResizeBehaviorSpec.js:328]: Style nit-pick: Please separate specs (`it(...)`) from each other with two empty lines.

**Review by nikku — COMMENTED** [review]
Unrelated to this PR, another improvement possibility: Lanes currently "snap during resize" with other elements in a participant. We should disallow it:

capture A8sQ5h_optimized

Other than that this works as expected, UX wise.

**Review by nikku — CHANGES_REQUESTED** [review]
Please clarify https://github.com/bpmn-io/bpmn-js/pull/2218#discussion_r1734442678 and update code style (tests).

File: lib/features/modeling/behavior/ResizeBehavior.js (hunk 3 above)
Comment (abdul99ahad) [reply lib/features/modeling/behavior/ResizeBehavior.js:172]: Yes, good catch. I'll update it.

File: test/spec/features/modeling/behavior/ResizeBehaviorSpec.js (hunk 4 above)
Comment (abdul99ahad) [reply test/spec/features/modeling/behavior/ResizeBehaviorSpec.js:328]: Updated. [q] Can we use es-lint custom rules to enforce this?

Comment (nikku) [reply test/spec/features/modeling/behavior/ResizeBehaviorSpec.js:328]: > [q] Can we use es-lint custom rules to enforce this?

Yes. You could write one, and contribute to [eslint-plugin-bpmn-io](https://github.com/bpmn-io/eslint-plugin-bpmn-io).

**Review by nikku — APPROVED** [review]

**Pull request merged**